source/common/event/user_space_file_event_impl.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <cstdint>
+
+#include "envoy/event/file_event.h"
+
+#include "common/event/dispatcher_impl.h"
+#include "common/event/event_impl_base.h"
+
+namespace Envoy {
+
+namespace Network {
+class BufferedIoSocketHandleImpl;
+}
+namespace Event {
+
+// Forward declare for friend class.
+class UserSpaceFileEventFactory;
+
+// The interface of populating event watcher and obtaining the active events. The events are the
+// combination of FileReadyType values. The event listener is populated by user event registration
+// and io events passively. Also the owner of this listener query the activated events by calling
+// triggeredEvents and getAndClearEphemeralEvents.
+class EventListener {
+public:
+  virtual ~EventListener() = default;
+
+  // Get the events which are enabled and triggered.
+  virtual uint32_t triggeredEvents() PURE;
+  // Get the events which are ephemerally activated. Upon returning the ephemeral events are
+  // cleared.
+  virtual uint32_t getAndClearEphemeralEvents() PURE;
+
+  /**
+   * FileEvent::setEnabled is invoked.
+   * @param enabled_events supplied the event of setEnabled.
+   */
+  virtual void onEventEnabled(uint32_t enabled_events) PURE;
+
+  /**
+   * FileEvent::activate is invoked.
+   * @param enabled_events supplied the event of activate().
+   */
+  virtual void onEventActivated(uint32_t activated_events) PURE;
+};
+
+// Return the enabled events except EV_CLOSED. This implementation is generally good since only
+// epoll supports EV_CLOSED but the entire envoy code base supports another poller. The event owner
+// must assume EV_CLOSED is never activated. Also event owner must tolerate that OS could notify
+// events which are not actually triggered.
+// TODO(lambdai): Add support of delivering EV_CLOSED.
+class DefaultEventListener : public EventListener {
```

Comment:
Please change to EventListenerImpl

I don't see there being multiple implementations of this event listener.

Reply:
:+1: